src/parser.rs

```diff
@@ -984,6 +984,25 @@ impl<'a> Parser<'a> {
         }
     }
 
+    pub fn parse_not(&mut self) -> Result<Expr, ParserError> {
+        match self.peek_token() {
+            Token::Word(w) => match w.keyword {
+                Keyword::EXISTS => {
+                    let negated = self.parse_keyword(Keyword::EXISTS);
```

Comment:
don't we know here that `negated` is true? I think the code might be clearer if this were something more like

```rust
let negated = true;
self.parse_keyword(Keyword::Exists);
self.parse_exists_expr(negated)
```

What do you think?